presto-main/src/test/java/com/facebook/presto/execution/buffer/TestArbitraryOutputBuffer.java

```diff
@@ -565,7 +565,6 @@ public void testFinishFreesReader()
         assertFalse(future.isDone());
 
         // finish the buffer
-        buffer.setNoMorePages();
```

Comment:
I'm temporarily taking this over for @arhimondr -- what do you mean by a test that wakes on `setNoMorePages`?